[release] v9.3.0 (#22540)
mui/mui-x: CHANGELOG.md

# Changelog

## 9.3.0

_May 21, 2026_

> We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

Reviewer: ```suggestion
We'd like to extend a big thank you to the 16 contributors who made this release possible. Here are some highlights ✨:
```

Wow that's a lot

- 📆 Refactor Calendar Range drag editing to use Pointer Events instead of drag and touch events combination (#22279)
- ✨ Apply lazy loading to `EventTimelinePremium` (#22308)
- 🐞 Bugfixes
- 📚 Documentation improvements
- 🌎 Added Norwegian bokmål (nb-NO) locale to the Scheduler
- 🌎 Improved Swedish (sv-SE) locale on the Data Grid

Special thanks go out to these community members for their valuable contributions:

> @Anexus5919, @jvskriubakken, @rin, @viktormelin

Reviewer: ```suggestion
@Anexus5919, @jvskriubakken, @oscar-b, @rin, @viktormelin
```

Same changes as in `@mui/x-data-grid@9.3.0`.

#### `@mui/x-data-grid-premium@9.3.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@9.3.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@9.3.0`

- [pickers] Fix disabled state styling for `PickersTextField` standard and filled variants (#22189) @LukasTy
- [pickers] Fix previous month disabled state calculation (#22524) @mj12albert
- [pickers] Merge `slotProps.sectionContent` in `PickersInputBase` (#22324) @rin

#### `@mui/x-date-pickers-pro@9.3.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@9.3.0`, plus:

- [DateRangeCalendar] Use Pointer Events for drag editing (#22279) @LukasTy

### Charts

#### `@mui/x-charts@9.3.0`

- [charts] Add `data-series` to elements of radar chart (#22523) @alexfauquette
- [charts] Add environment check for warning messages in axis and radar hooks (#22506) @sai6855
- [charts] Chart `seriesConfig` deduplication for lines and bars (#22257) @sai6855

> - [charts] Fix the line and mark class (#22526) @alexfauquette

Reviewer: ```suggestion
- [charts] Fix highlighted item crash (#22539) @oscar-b
- [charts] Fix the line and mark class (#22526) @alexfauquette
```

# Changelog

## 9.2.1

_May 21, 2026_

We'd like to extend a big thank you to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 🐞 Bugfixes

Reviewer 2: ```suggestion
- 📆 Refactor Calendar Range drag editing to use Pointer Events instead of drag and touch events combination (#22279)
- 🐞 Bugfixes
```

> - 📆 Refactor Calendar Range drag editing to use Pointer Events instead of drag and touch events combination (#22279)

Reviewer 3: ```suggestion
- 📆 Refactor Calendar Range drag editing to use Pointer Events instead of drag and touch events combination (#22279)
- ✨ Apply lazy loading to `EventTimelinePremium` (#22308)
```
This is a huge/premium feature, but not sure if we want to add something in alpha as a highlight 👀

Reviewer 4: I think we do 👌

- 📚 Documentation improvements
- 🌎 Added Norwegian bokmål (nb-NO) locale to the Scheduler

> - 🌎 Improved Swedish (svSE) locale on the Data Grid

Reviewer 5: ```suggestion
- 🌎 Improved Swedish (sv-SE) locale on the Data Grid
```

### Data Grid

#### `@mui/x-data-grid@9.2.1`

- [DataGrid] Fix scrollbar disappearing after multiple resizes (#22512) @LukasTy
- [DataGrid] Revert "Add support for pinned columns in `GridVirtualScroller`" (#22467) @oliviertassinari
- [l10n] Improve Swedish (svSE) locale (#21851) @viktormelin

#### `@mui/x-data-grid-pro@9.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@9.2.1`.

#### `@mui/x-data-grid-premium@9.2.1` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@9.2.1`.

### Date and Time Pickers

#### `@mui/x-date-pickers@9.2.1`

- [pickers] Fix disabled state styling for `PickersTextField` standard and filled variants (#22189) @LukasTy
- [pickers] Fix previous month disabled state calculation (#22524) @mj12albert
- [pickers] Merge `slotProps.sectionContent` in `PickersInputBase` (#22324) @rin

#### `@mui/x-date-pickers-pro@9.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@9.2.1`, plus:

> - [DateRangeCalendar] Use Pointer Events for drag editing (#22279) @LukasTy

Reviewer 2: This is a relatively noticeable and substantial refactor that could have certain unforeseen consequences on exotic use-cases.
I'd prefer a minor release if you don't mind.

- [charts] Fix use of deprecated API in docs (#22469) @oliviertassinari
- [charts] Improve performance of `evaluateCurveY` function (#22407) @sai6855

#### `@mui/x-charts-pro@9.2.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@9.2.1`.

#### `@mui/x-charts-premium@9.2.1` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-charts-pro@9.2.1`, plus:

- [charts-premium] Add `closePath` option to the radial line series (#22517) @alexfauquette

### Tree View

#### `@mui/x-tree-view@9.1.0`

Internal changes.

#### `@mui/x-tree-view-pro@9.1.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@9.1.0`.

### Scheduler

#### `@mui/x-scheduler@9.0.0-alpha.7`

- [scheduler] Apply lazy loading to `EventTimelinePremium` (#22308) @rita-codes
- [scheduler] Deduplicate staged ranges within the debounce window (#22476) @Anexus5919
- [scheduler] Moving recurring events to premium (#22430) @rita-codes
- [scheduler] Pass full event objects to `dataSource.updateEvents` (#22462) @rita-codes

> - [I10n] Add Norwegian bokmål (nb-NO) locale (#22415) @jvskriubakken

Reviewer 3: ```suggestion
- [I10n] Add Norwegian bokmål (nb-NO) locale (#22415) @jvskriubakken
```

#### `@mui/x-scheduler-premium@9.0.0-alpha.7` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-scheduler@9.0.0-alpha.7`.

> 

Reviewer 3: ```suggestion
Same changes as in `@mui/x-scheduler@9.0.0-alpha.7`, plus:

- [scheduler] Apply lazy loading to `EventTimelinePremium` (#22308) @rita-codes
- [scheduler] Deduplicate staged ranges within the debounce window (#22476) @Anexus5919
- [scheduler] Pass full event objects to `dataSource.updateEvents` (#22462) @rita-codes
```
